internal/servers/server/option_test.go

```diff
@@ -0,0 +1,2316 @@
+//
+// Copyright (C) 2019-2020 Vdaas.org Vald team ( kpango, rinx, kmrmt )
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+package server
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/vdaas/vald/internal/errgroup"
+	"github.com/vdaas/vald/internal/errors"
+	"google.golang.org/grpc"
+)
+
+func TestWithHost(t *testing.T) {
+	type test struct {
+		name      string
+		host      string
+		checkFunc func(opt Option) error
+	}
+
+	tests := []test{
+		{
+			name: "set success",
+			host: "host",
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.host != "host" {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "not set nothing when host is empty",
+			checkFunc: func(opt Option) error {
+				got := &server{
+					host: "host",
+				}
+				opt(got)
+
+				if got.host != "host" {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opt := WithHost(tt.host)
+			if err := tt.checkFunc(opt); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestWithPort(t *testing.T) {
+	type test struct {
+		name      string
+		port      uint
+		checkFunc func(opt Option) error
+	}
+
+	tests := []test{
+		{
+			name: "set success",
+			port: 8080,
+			checkFunc: func(opt Option) error {
+				got := new(server)
+				opt(got)
+
+				if got.port != 8080 {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+
+		{
+			name: "not set when port is 0",
+			checkFunc: func(opt Option) error {
+				got := &server{
+					port: 8080,
+				}
+				opt(got)
+
+				if got.port != 8080 {
+					return errors.New("invalid param was set")
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			opt := WithPort(tt.port)
+			if err := tt.checkFunc(opt); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
+func TestWithName(t *testing.T) {
+	type test struct {
+		name      string
+		n         string
```

Comment:
please use more meanful naming

Reply:
Thank you for your feedback.
I fixed about this contents.